engine/src/host_imports/ivalues_lowering.rs

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright 2020 Fluence Labs Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+use super::WValue;
+use super::AllocateFunc;
+use super::utils::write_to_wasm_mem;
+use crate::call_wasm_func;
+use crate::IValue;
+
+use wasmer_core::vm::Ctx;
+use wasmer_wit::vec1::Vec1;
+
```

Comment:
It would be nice to have a small comment on the meaning of the whole file, maybe on what "lowering" means in this context.